core/src/main/java/com/google/instrumentation/trace/SpanBuilder.java

```diff
@@ -0,0 +1,277 @@
+   * @return this.
+   */
+  public SpanBuilder setSampler(@Nullable Sampler sampler) {
+    startSpanOption.setSampler(sampler);
+    return this;
+  }
+
+  /**
+   * Sets the {@code List} of parent links. Links are used to link {@link Span}s in different
+   * traces. Used (for example) in batching operations, where a single batch handler processes
+   * multiple requests from different traces.
+   *
+   * @param parentLinks New links to be added.
+   * @return this.
+   */
+  public SpanBuilder setParentLinks(@Nullable List<Span> parentLinks) {
+    startSpanOption.setParentLinks(parentLinks);
+    return this;
+  }
+
+  /**
+   * Sets recordEvents.
+   *
+   * @param recordEvents New value determining if this {@code Span} should have events recorded. If
+   *     a {@code null} value is passed, the implementation will provide a default.
+   * @return this.
+   */
+  public SpanBuilder setRecordEvents(@Nullable Boolean recordEvents) {
+    startSpanOption.setRecordEvents(recordEvents);
+    return this;
+  }
+
+  /**
+   * If called this will force the newly created {@code Span} to be a root span. As a consequence,
+   * any parent specified (or inherited from the Context) will be ignored (N.B. does not apply to
+   * linked parents set through {@link #setParentLinks}).
+   *
+   * @return this.
+   */
+  public SpanBuilder becomeRoot() {
+    parentSpanContext = null;
+    hasRemoteParent = false;
+    return this;
+  }
+
+  /**
+   * Creates and starts a new {@link Span}.
+   *
+   * <p>If used without try-with-resources <b>must</b> manually call {@link Span#end}.
+   *
+   * <p>Does not install the newly created {@code Span} to the current Context.
+   *
+   * <p>Example of usage:
+   *
+   * <pre>{@code
+   * class MyClass {
+   *   private static final Tracer tracer = Tracer.getTracer();
+   *   void DoWork() {
+   *     try (Span span = tracer.spanBuilder(null, "MyRootSpan").startSpan()) {
+   *       span.addAnnotation("We did X");
+   *     }
+   *   }
+   * }
+   * }</pre>
+   *
+   * <p>Prior to Java SE 7, you can use a finally block to ensure that a resource is closed
+   * regardless of whether the try statement completes normally or abruptly.
+   *
+   * <p>Example of usage prior to Java SE7:
+   *
+   * <pre>{@code
+   * class MyClass {
+   *   private static final Tracer tracer = Tracer.getTracer();
+   *   void DoWork() {
+   *     Span span = tracer.spanBuilder(null, "MyRootSpan").startSpan();
+   *     try {
+   *       span.addAnnotation("We did X");
+   *     } finally {
+   *       span.close();
+   *     }
+   *   }
+   *  }
+   * }</pre>
+   *
+   * @return the newly created {@code Span}.
+   */
+  public Span startSpan() {
+    return start();
+  }
+
+  // TODO(bdrutu): Add error_prone annotation @MustBeClosed when the 2.0.16 jar is fixed.
+  /**
+   * Creates and starts a new {@link Span} and returns an object that defines a scope where the
```

Comment:
I'd simplify this comment and move the more literal one below the example. It would make it more easily accessible.

```
Starts a new new span and sets it as the {@link Tracer#getCurrentSpan current span}.
```